src/services/api/scheduling.ts

```diff
@@ -102,6 +102,80 @@ function generateSchedulingConfiguration({
   })
 }
 
+/**
+ * Create a Nylas Session
+ *
+ * #### Example
+ *
+ * ```js
+ * const { sessionId, configurationId, bookingRef, name, email } =
+ *   await schedulingService.createNylasSession(
+ *     '89c6e98e-f56f-45fc-84fe-c4fc62331d34',
+ *   )
+ * // use sessionId and configurationId/bookingRef to create or modify nylas bookings
+ * ```
+ *
+ * @param submissionId
+ * @returns
+ */
+function createNylasSession(submissionId: string, abortSignal?: AbortSignal) {
+  const url = `${tenants.current.apiOrigin}/nylas/authorise-booking`
+  return postRequest<{
+    sessionId: string
+    configurationId: string
+    name: string | undefined
+    email: string | undefined
+    bookingRef: string | undefined
+  }>(url, { submissionId }, abortSignal).catch((error) => {
+    Sentry.captureException(error)
+    console.warn(
+      'Error occurred while attempting to create a nylas session',
+      error,
+    )
+    switch (error.status) {
+      case 401: {
+        throw new OneBlinkAppsError(
+          'You cannot start a session until you have logged in. Please login and try again.',
+          {
+            originalError: error,
+            httpStatusCode: error.status,
+            requiresLogin: true,
+          },
+        )
+      }
+      case 403: {
+        throw new OneBlinkAppsError(
+          'You may not create or modify a calendar booking for a form you did not submit',
+          {
+            originalError: error,
+            httpStatusCode: error.status,
+            requiresAccessRequest: true,
+          },
+        )
+      }
+      case 400:
```

Comment:
Can we please have a case for 400 where we pass the `error.message` as the message.